cdap-tms/src/main/java/co/cask/cdap/messaging/subscriber/AbstractMessagingSubscriberService.java

```diff
@@ -159,58 +132,26 @@ protected final String loadMessageId() {
   @Nullable
   @Override
   protected String processMessages(Iterator<ImmutablePair<String, T>> messages) {
-    int curTxTimeout = txTimeoutSeconds;
     MessageTrackingIterator iterator;
 
     while (true) {
-      try {
-        // Process the notifications and record the message id of where the processing is up to.
-        // 90% of the tx timeout is .9 * 1000 * txTimeoutSeconds = 900 * txTimeoutSeconds
-        long timeBoundMillis = 900L * curTxTimeout;
-        // TODO: CDAP-14848 Remove redundancy
-        iterator = Transactionals.execute(getTransactional(), curTxTimeout, context ->
-          TransactionRunners.run(getTransactionRunner(), context1 -> {
-            TimeBoundIterator<ImmutablePair<String, T>> timeBoundMessages = new TimeBoundIterator<>(messages,
```

Comment:
is it safe to remove the TimeboundIterator? When the StructuredTables are backed by Datasets, we still have transactions and will encounter transaction timeouts - only that they are happening in the TransactionRunner.

Reply:
It is only used in the `AbstractMessagingSubcriberService`. I believe we add this class as a temporary mediation for the timeout to process messages. I can keep it in case we want to use it in future

Reply:
If these timeouts cannot happen any more, then it's fine to remove this iterator. But if the timeouts can still happen. it would be better to keep it

Reply:
Good point, will add it back.